llama_stack/providers/remote/vector_io/milvus/milvus.py

```diff
@@ -54,24 +53,27 @@ def sanitize_collection_name(name: str) -> str:
 
 class MilvusIndex(EmbeddingIndex):
     def __init__(
-        self, client: MilvusClient, collection_name: str, consistency_level="Strong", kvstore: KVStore | None = None
+        self,
+        client: AsyncMilvusClient,
+        collection_name: str,
+        consistency_level="Strong",
+        kvstore: KVStore | None = None,
     ):
         self.client = client
         self.collection_name = sanitize_collection_name(collection_name)
         self.consistency_level = consistency_level
         self.kvstore = kvstore
 
     async def delete(self):
-        if await asyncio.to_thread(self.client.has_collection, self.collection_name):
-            await asyncio.to_thread(self.client.drop_collection, collection_name=self.collection_name)
+        if await self.client.collection_exists(self.collection_name):
```

Comment:
Why did you change these from `has_collection` to `collection_exists`? That is at least part of your test failures. It's incorrect, but I'm curious as to why you chose to change it. Is there some version of the async client that had this `collection_exists` method?

Reply:
so the failure in CI is caused because async client does not have has_collection iiuc

Reply:
Hmm - `has_collection` is not actually in any of the released versions of AsyncMilvusClient yet. Do we really want to migrate to the async client, given how new some of the methods we use are and this warning at the top of that source code?

https://github.com/milvus-io/pymilvus/blob/9e95be63b3d4697c53579efee86d839a1de9f926/pymilvus/milvus_client/async_milvus_client.py#L33-L34

```python
    """AsyncMilvusClient is an EXPERIMENTAL class
    which only provides part of MilvusClient's methods"""
```